ngx_http_pipelog_module.c

```diff
@@ -2209,6 +2231,12 @@ ngx_http_pipelog_logger_process_main (ngx_cycle_t *cycle) {
     while (1) {
         sig = sigtimedwait(&set, NULL, &timeout);
         ngx_time_update();
+
+        if(sig == SIGTERM) {
+            ngx_log_error(NGX_LOG_ALERT, cycle->log, 0, "%s: ngx_http_pipelog_command_exec(): SIGTERM detected, gracefully shutting down...", MODULE_NAME);
+            _exit(0);
+        }
```

Comment:
If you kill() a process group here, the processes under it will be terminated.

Reply:
I added _exit(0) so that the child processes would automatically get SIGKILL via the PR_SET_PDEATSIG

Reply:
In that case, probably only the child process `bash` will exit and the grandchild process `cat` command will leak.

Reply:
It looks like that bash doesn't do anything w/ SIGTERM by default for a shell. Looks like I will have to just do SIGKILL. Or perhaps a config option to set what signals it sends to it's piped logger processes? 

For our codebase, we use a C program that does the piped logging - it has signal handlers in them and this code works well. I guess it doesn't work if a piped logger program does not have signal handling.